apache-solr/dashboards/apache-solr-cluster-overview.libsonnet

```diff
@@ -0,0 +1,1680 @@
+            color: 'red',
+            value: 80,
+          },
+        ],
+      },
+      unit: 'percent',
+    },
+    overrides: [],
+  },
+  options: {
+    legend: {
+      calcs: [],
+      displayMode: 'list',
+      placement: 'right',
+      showLegend: true,
+    },
+    tooltip: {
+      mode: 'multi',
+      sort: 'desc',
+    },
+  },
+};
+
+local topHeapMemoryUsageByNodePanel = {
+  datasource: promDatasource,
+  targets: [
+    prometheus.target(
+      'topk($k, 100 * avg by (job, base_url, solr_cluster) (sum without(item)(solr_metrics_jvm_memory_heap_bytes{job=~"$job", solr_cluster=~"$solr_cluster", item="used"}) / clamp_min(sum without(item)(solr_metrics_jvm_memory_heap_bytes{job=~"$job", solr_cluster=~"$solr_cluster", item="max"}), 1)))',
+      datasource=promDatasource,
+      legendFormat='{{base_url}}',
+      format='time_series',
+    ),
+  ],
+  type: 'timeseries',
+  title: 'Top heap memory usage by node',
+  description: 'Top JVM heap memory usage.',
+  fieldConfig: {
+    defaults: {
+      color: {
+        mode: 'continuous-BlYlRd',
+      },
+      custom: {
+        axisCenteredZero: false,
+        axisColorMode: 'text',
+        axisLabel: '',
+        axisPlacement: 'auto',
+        barAlignment: 0,
+        drawStyle: 'line',
+        fillOpacity: 30,
+        gradientMode: 'none',
+        hideFrom: {
+          legend: false,
+          tooltip: false,
+          viz: false,
+        },
+        lineInterpolation: 'smooth',
+        lineWidth: 2,
+        pointSize: 5,
+        scaleDistribution: {
+          type: 'linear',
+        },
+        showPoints: 'never',
+        spanNulls: false,
+        stacking: {
+          group: 'A',
+          mode: 'none',
+        },
+        thresholdsStyle: {
+          mode: 'off',
+        },
+      },
+      mappings: [],
+      min: 0,
+      max: 100,
+      thresholds: {
+        mode: 'absolute',
+        steps: [
+          {
+            color: 'blue',
+          },
+          {
+            color: 'yellow',
+            value: 90,
+          },
+          {
+            color: 'red',
+            value: 80,
+          },
+        ],
+      },
+      unit: 'percent',
+    },
+    overrides: [],
+  },
+  options: {
+    legend: {
+      calcs: [],
+      displayMode: 'list',
+      placement: 'right',
+      showLegend: true,
+    },
+    tooltip: {
+      mode: 'multi',
+      sort: 'desc',
+    },
+  },
+};
+
+local topMeanQueriesByNodePanel = {
+  datasource: promDatasource,
+  targets: [
+    prometheus.target(
+      'topk($k, avg by(job, base_url, solr_cluster, collection, core, searchHandler) (solr_metrics_core_query_mean_rate{job=~"$job", solr_cluster=~"$solr_cluster", category="QUERY", collection=~"$solr_collection", core=~"$solr_core"}))',
+      datasource=promDatasource,
+      legendFormat='{{collection}} - {{core}} - {{searchHandler}}',
+      format='time_series',
+    ),
+  ],
+  type: 'timeseries',
+  title: 'Top mean queries by core',
```

Comment:
For all the top x by y titles we have, wouldnt it be better as top y by x? For example "Top mean queries by core" -> "Top cores by mean queries". Im not sure but I think it sounds better this way? Especially for some of the later ones like "top core by index size".

Reply:
I don't think so. Having the most important details that makes the panel unique first, and then the selector at the end makes the more sense to me.

Reply:
Bomin is right that we've historically titled these with the selector first. I see both ways here, but I think what we've been doing makes the most sense.

Reply:
I see, I'll update the titles then